packages/web3-account-abstraction/src/web3_aa.ts

```diff
@@ -0,0 +1,340 @@
+	 *
+	 * @param UserOperation - represents the structure of a transaction initiated by the user. It contains the sender, receiver, call data, maximum fee per unit of Gas, maximum priority fee, signature, nonce, and other specific elements.
+	 * @param entryPoint - a singleton contract to execute bundles of UserOperations. Bundlers/Clients whitelist the supported entrypoint.
+	 * @returns - `preVerificationGas` gas overhead of this UserOperation
+	 * - `verificationGasLimit` actual gas used by the validation of this UserOperation
+	 * - `callGasLimit` value used by inner account execution
+	 * @example
+	 * ```ts
+	 * 	estimateUserOperationGas({
+	 *		sender: "0x9fd042a18e90ce326073fa70f111dc9d798d9a52",
+	 *		nonce: "123",
+	 *		initCode: "0x68656c6c6f",
+	 *		callData: "0x776F726C64",
+	 *		callGasLimit: "1000",
+	 *		verificationGasLimit: "2300",
+	 *		preVerificationGas: "3100",
+	 *		maxFeePerGas: "0",
+	 *		maxPriorityFeePerGas: "0",
+	 *		paymasterAndData: "0x626c6f63746f",
+	 *		signature: "0x636c656d656e74"
+	 *	},"0x636c656d656e74").then(console.log);
+	 * > {
+	 * 		callGasLimit : "0x18b33",
+	 *		preVerificationGas: "0xdf17",
+	 *		verificationGasLimit:"0x128c4"
+	 *	 }
+	 * ```
+	 */
+	public async estimateUserOperationGas(userOperation: UserOperation, entryPoint: Address) {
+		const userOp = { ...userOperation };
+		if (userOperation?.maxFeePerGas === undefined) {
+			userOp.maxFeePerGas = '0';
+		}
+		if (userOperation?.maxPriorityFeePerGas === undefined) {
+			userOp.maxPriorityFeePerGas = '0';
+		}
+		return (this.bundlerRequestManager ?? this.requestManager).send({
+			method: 'eth_estimateUserOperationGas',
+			params: [userOp, entryPoint],
+		});
+	}
+	/**
+	 * Return a UserOperation based on a hash (userOpHash) returned by eth_sendUserOperation
+	 *
+	 * @param hash - a userOpHash value returned by `eth_sendUserOperation`
+	 * @returns null in case the UserOperation is not yet included in a block, or a full UserOperation, with the addition of entryPoint, blockNumber, blockHash and transactionHash
+	 * @example
+	 * ```ts
+	 * 	getUserOperationByHash("0xxxxx").then(console.log);
+	 * > {
+	 *  userOperation: {
+	 *		sender: "xxxx",
+	 *		nonce: "xxxx",
+	 *		initCode: "0xxxxxxxxxxxxxxx",
+	 *		callData: "0xxxxxxxxxxxxxxx",
+	 *		callGasLimit: "xxxx",
+	 *		verificationGasLimit: "xxxx",
+	 *		preVerificationGas: "xxxx",
+	 *		maxFeePerGas: "0",
+	 *		maxPriorityFeePerGas: "0",
+	 *		paymasterAndData: "0xxxxxxxxxxxxxxx",
+	 *		signature: "0xxxxxxxxxxxxxxx"
+	 *	},
+	 *	entryPoint: '0x5ff137d4b0fdcd49dca30c7cf57e578a026d2789',
+	 *	blockNumber: 39642225,
+	 *	blockHash: '0x638a175940cacb33f35e265961b164b14aa77477438340e635b227752f31981f',
+	 *	transactionHash: '0xc3e64ac247ae2343335596e106d1b97e35637656e1b99b37977a4165b34aeeb4',
+	 * }
+	 * ```
+	 */
+
+	public async getUserOperationByHash(hash: HexStringBytes) {
```

Comment:
I know this is the first version, but it would be good to validate params before each request in the future. Also, it will be great to have a return format param and format the result for each request

Reply:
added that in https://github.com/web3/web3.js/issues/7202 list